systest/tests/distributed_post_verification_test.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/spacemeshos/go-spacemesh/sql/localsql"
+	"github.com/spacemeshos/go-spacemesh/systest/cluster"
+	"github.com/spacemeshos/go-spacemesh/systest/testcontext"
+	"github.com/spacemeshos/go-spacemesh/timesync"
+	"github.com/spacemeshos/go-spacemesh/timesync/peersync"
+)
+
+var grpclog grpc_logsettable.SettableLoggerV2
+
+func init() {
+	grpclog = grpc_logsettable.ReplaceGrpcLoggerV2()
+}
+
+func TestPostMalfeasanceProof(t *testing.T) {
+	t.Parallel()
+	testDir := t.TempDir()
+
+	ctx := testcontext.New(t, testcontext.Labels("sanity"))
+	logger := ctx.Log.Desugar().WithOptions(zap.IncreaseLevel(zapcore.InfoLevel), zap.WithCaller(false))
+
+	// Prepare cluster
+	ctx.PoetSize = 1 // one poet guarantees everybody gets the same proof
+	ctx.ClusterSize = 3
+	cl := cluster.New(ctx, cluster.WithKeys(10))
+	require.NoError(t, cl.AddBootnodes(ctx, 1))
+	require.NoError(t, cl.AddBootstrappers(ctx))
+	require.NoError(t, cl.AddPoets(ctx))
+	require.NoError(t, cl.AddSmeshers(ctx, ctx.ClusterSize-cl.Total(), cluster.WithFlags(cluster.PostK3(1))))
+
+	// Prepare config
+	cfg, err := presets.Get("fastnet")
+	require.NoError(t, err)
+	cfg.Genesis = &config.GenesisConfig{
+		GenesisTime: cl.Genesis().Format(time.RFC3339),
+		ExtraData:   cl.GenesisExtraData(),
+	}
+	cfg.LayersPerEpoch = uint32(testcontext.LayersPerEpoch.Get(ctx.Parameters))
+	types.SetLayersPerEpoch(cfg.LayersPerEpoch)
+	cfg.LayerDuration = testcontext.LayerDuration.Get(ctx.Parameters)
+
+	cfg.DataDirParent = testDir
+	cfg.SMESHING.Opts.DataDir = filepath.Join(testDir, "post-data")
+	cfg.P2P.DataDir = filepath.Join(testDir, "post-data")
+	require.NoError(t, os.Mkdir(cfg.P2P.DataDir, os.ModePerm))
+
+	cfg.PoetServers = []types.PoetServer{
+		{Address: cluster.MakePoetGlobalEndpoint(ctx.Namespace, 0)},
+	}
+
+	var bootnodes []*cluster.NodeClient
+	for i := 0; i < cl.Bootnodes(); i++ {
+		bootnodes = append(bootnodes, cl.Client(i))
+	}
+
+	endpoints, err := cluster.ExtractP2PEndpoints(ctx, bootnodes)
+	require.NoError(t, err)
+	cfg.P2P.Bootnodes = endpoints
+	cfg.P2P.PrivateNetwork = true
+	cfg.Bootstrap.URL = cluster.BootstrapperGlobalEndpoint(ctx.Namespace, 0)
+	cfg.P2P.MinPeers = 2
+	ctx.Log.Infow("Prepared config", "cfg", cfg)
+
+	goldenATXID := cl.GoldenATX()
+	signer, err := signing.NewEdSigner(signing.WithPrefix(cl.GenesisID().Bytes()))
+	require.NoError(t, err)
+
+	prologue := fmt.Sprintf("%x-%v", cl.GenesisID(), cfg.LayersPerEpoch*2-1)
+	host, err := p2p.New(
+		ctx,
+		log.NewFromLog(logger.Named("p2p")),
+		cfg.P2P,
+		[]byte(prologue),
+		handshake.NetworkCookie(prologue),
+	)
+	require.NoError(t, err)
+	logger.Info("p2p host created", zap.Stringer("id", host.ID()))
+	host.Register(pubsub.AtxProtocol, func(context.Context, peer.ID, []byte) error { return nil })
+	ptimesync := peersync.New(
+		host,
+		host,
+		peersync.WithLog(log.NewFromLog(logger.Named("peersync"))),
+		peersync.WithConfig(cfg.TIME.Peersync),
+	)
+	ptimesync.Start()
+	t.Cleanup(ptimesync.Stop)
```

Comment:
What is `ptimesync` required for in this test?

Reply:
:thinking: It's not needed, thanks